accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -50,19 +50,21 @@ More precisely, an extension declaration is a top-level declaration with a gramm
 
 ```ebnf
 <extension> ::= 
-  `extension' <identifier>? <typeParameters>? `on' <type> `?'? `{'
+  `extension' <identifier>? <typeParameters>? `on' <type> `{'
      memberDeclaration*
   `}'
 ```
 
 Such a declaration introduces its *name* (the identifier) into the surrounding scope. The name does not denote a type, but it can be used to denote the extension itself in various places. The name can be hidden or shown in `import` or `export` declarations.
 
-The *type* can be any valid Dart type, including a single type variable. It can refer to the type parameters of the extension. It can be followed by `?` which means that it allows `null` values. When Dart gets non-nullable types by default (NNBD), this `?` syntax is removed and subsumed by nullable types like `int?` being allowed in the `<type>` position.
+The *type* can be any valid Dart type, including a single type variable. It can refer to the type parameters of the extension.
 
 The member declarations can be any non-abstract static or instance member declaration except for instance variables and constructors. Instance member declaration parameters must not be marked `covariant`. Abstract members are not allowed since the extension declaration does not introduce an interface, and constructors are not allowed because the extension declaration doesn't introduce any type that can be constructed. Instance variables are not allowed because there won't be any memory allocation per instance that the extension applies to. We could implement instance variables using an `Expando`, but it would necessarily be nullable, so it would still not be an actual instance variable.
 
 An extension declaration with a non-private name is included in the library's export scope, and a privately named extension is not. It is a compile-time error to export two declarations, including extensions, with the same name, whether they come from declarations in the library itself or from export declarations (with the usual exception when all but one declaration come from platform libraries). Extension *members* with private names are simply inaccessible in other libraries.
 
+We may want to make `extension` a built-in identifier. Is not necessary for disambiguation, but it may make parsing easier.
```

Comment:
Do we or don't we?

Reply:
Yes, `extension` is currently a built-in identifier and allows me to implement better parser recovery... especially when braces are mismatched.

Reply:
Is `on` also a built-in identifier, or was that not necessary?